pkg/operator/ceph/cluster/mgr/spec.go

```diff
@@ -135,23 +135,29 @@ func (c *Cluster) needHttpBindFix() bool {
 // always do this and (b) make sure that all forms of the configuration option
 // are removed (see the init container factory method). Once the minimum
 // supported version of Rook contains the fix, all of this can be removed.
-func (c *Cluster) clearHttpBindFix(mgrConfig *mgrConfig) {
-	for _, module := range []string{"dashboard", "prometheus"} {
-		// there are two forms of the configuration key that might exist which
-		// depends not on the current version, but on the version that may be
-		// the version being upgraded from.
-		for _, ver := range []cephver.CephVersion{cephver.Mimic} {
-			changed, err := client.MgrSetConfig(c.context, c.Namespace, mgrConfig.DaemonID, ver,
-				fmt.Sprintf("mgr/%s/server_addr", module), "", false)
-			logger.Infof("clearing http bind fix mod=%s ver=%s changed=%t err=%+v", module, &ver, changed, err)
-
-			// this is for the format used in v1.0
-			// https://github.com/rook/rook/commit/11d318fb2f77a6ac9a8f2b9be42c826d3b4a93c3
-			changed, err = client.MgrSetConfig(c.context, c.Namespace, mgrConfig.DaemonID, ver,
-				fmt.Sprintf("mgr/%s/%s/server_addr", module, mgrConfig.DaemonID), "", false)
-			logger.Infof("clearing http bind fix mod=%s ver=%s changed=%t err=%+v", module, &ver, changed, err)
+func (c *Cluster) clearHTTPBindFix() error {
+	// We only need to apply these changes once. No harm in once each time the operator restarts.
+	if c.appliedHttpBind {
+		return nil
+	}
+	for _, daemonID := range c.getDaemonIDs() {
```

Comment:
Can we do away with this loop if we enable mgr modules for all mgrs instead of individually?

Reply:
Seems like we needed these to be configured on individual daemons. @noahdesu do you recall? In any case, this code just moved around so it is equivalent to what we had before.

Reply:
hey gang! the only thing that comes to mind here is that, while it seems as though a configuration should be able to be set generically (e.g. `mgr/<module>/server_addr` which I think is what @BlaineEXE  is referring to) I recall that if one supports versions of ceph back far enough, there were weird inconsistencies in either (1) ceph configuration handling and/or (2) the way a particular module was querying for its configuration values.

ultimately for dashboard/prometheus i recall looking at each of the few supported versions and handling the cases explicitly. hope that helps!

Reply:
@noahdesu Thanks for the background! That does ring a bell now so I wouldn't want to change it, at least not as part of this PR.